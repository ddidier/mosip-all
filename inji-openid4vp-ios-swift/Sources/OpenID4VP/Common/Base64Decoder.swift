import Foundation

class Base64Decoder {
    static let className = String(describing: Base64Decoder.self)
    
    static func decodeBase64ToJSON(_ base64String: String) throws -> [String: Any] {
        let standardBase64String = makeBase64Standard(base64String)
        guard let decodedData = Data(base64Encoded: standardBase64String) else {
            throw Logger.handleException(exceptionType: "Decoding", message: "JWT payload decoding failed" ,className: Base64Decoder.className)
        }
        do{
            guard let jsonObject = try JSONSerialization.jsonObject(with: decodedData, options: []) as? [String: Any]  else {
                throw Logger.handleException(
                    exceptionType: "InvalidData",
                    message: "Decoding failed",
                    className: Base64Decoder.className
                )
            }
            return jsonObject
        } catch {
            throw Logger.handleException(exceptionType: "JsonDecodingFailed", message: "JWT Decoding to json failed", className: Base64Decoder.className)
        }
    }
    
    static func makeBase64Standard(_ base64String: String) -> String {
        var validBase64String = base64String
            .replacingOccurrences(of: "-", with: "+")
            .replacingOccurrences(of: "_", with: "/")
        
        while validBase64String.count % 4 != 0 {
            validBase64String.append("=")
        }
        return validBase64String
    }
}
