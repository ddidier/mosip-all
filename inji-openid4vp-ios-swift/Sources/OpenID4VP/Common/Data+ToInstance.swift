import Foundation

extension Data {
    func toInstance<T: Decodable>(as type: T.Type) throws -> T {
        return try JSONDecoder().decode(T.self, from: self)
    }
}
