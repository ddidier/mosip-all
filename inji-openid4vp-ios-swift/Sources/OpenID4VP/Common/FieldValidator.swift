import Foundation

func isNeitherNullNorEmpty(field: String) -> Bool {
    return field != "nil" && !field.isEmpty
}
