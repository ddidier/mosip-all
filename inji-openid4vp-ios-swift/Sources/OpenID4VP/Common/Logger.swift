import Foundation

class Logger {
    private static var logTag = ""
    private static var traceabilityId: String?
    
    static func setTraceabilityId(className: String, traceabilityId: String? = nil) {
        if let traceId = traceabilityId {
            self.traceabilityId = traceId
        }
    }
    static func getLogTag(_ className: String) -> String {
        return "INJI-OpenID4VP : \(className) | traceID \(String(describing: self.traceabilityId))"
    }
    
    static func error(_ logTag: String, _ exception: Error) {
        print("\(logTag) | ERROR: \(exception.localizedDescription)")
    }
    
    static func handleException(exceptionType: String, message: String? = nil, fieldPath: [String]? = nil, className: String) -> Error {
        var fieldPathAsString: String = ""
        if let fieldPath = fieldPath{
            fieldPathAsString = fieldPath.joined(separator: "->")
        }
        let exception: Error
        switch exceptionType {
        case "MissingInput":
            exception = AuthorizationRequestException.missingInput(fieldPath: fieldPathAsString)
        case "InvalidInput":
            exception = AuthorizationRequestException.invalidInput(fieldPath: fieldPathAsString)
        case "InvalidInputPattern":
            exception = AuthorizationRequestException.invalidInputPattern(fieldPath: fieldPathAsString)
        case "InvalidQueryParams":
            exception = AuthorizationRequestException.invalidQueryParams(message: message ?? "")
        case "UTF8Encoding":
            exception = AuthorizationRequestException.utf8Encoding(fieldPath: fieldPathAsString)
        case "JsonDecodingFailed":
            exception = AuthorizationRequestException.jsonDecodingFailed(fieldPath: fieldPathAsString, message: message ?? "")
        case "JsonEncodingFailed":
            exception = AuthorizationRequestException.jsonEncodingFailed(fieldPath: fieldPathAsString, message: message ?? "")
        case "Decoding":
            exception = AuthorizationRequestException.decodingException(fieldPath: fieldPathAsString)
        case "InvalidVerifier":
            exception = AuthorizationRequestException.invalidVerifier(message: message)
        case "MismatchingClientIDInRequest":
            exception = AuthorizationRequestException.mismatchingClientIDInRequest
        case "MismatchingClientIdSchemeInRequest":
            exception = AuthorizationRequestException.mismatchingClientIdSchemeInRequest
        case "InvalidVerifierRedirectUri":
            exception = AuthorizationRequestException.invalidVerifierRedirectUri
        case "InvalidLimitDisclosure":
            exception = AuthorizationRequestException.invalidLimitDisclosure
        case "InvalidClientIdScheme":
            exception = JwtVerificationException.invalidClientIdScheme(message: message ?? "")
        case "UrlCreationFailed":
            exception = NetworkRequestException.urlCreationFailed(message: message ?? "")
        case "PublicKeyNotFound":
            exception = JwtVerificationException.publicKeyNotFound(message: message)
        case "PublicKeyExtractionFailed":
            exception = JwtVerificationException.publicKeyExtractionFailed
        case "KidExtractionFailed":
            exception = JwtVerificationException.kidExtractionFailed(message: message ?? "")
        case "InvalidSignature":
            exception = JwtVerificationException.invalidSignature(message: message ?? "")
        case "ProofVerificationFailed":
            exception = JwtVerificationException.proofVerificationFailed(message: message ?? "")
        case "UnsupportedHttpMethod" :
            exception = AuthorizationRequestException.unsupportedHttpMethod(message: message ?? "")
        case "InvalidData":
            exception = AuthorizationRequestException.invalidData(message: message ?? "")
        case "InvalidResponseMode":
            exception = AuthorizationRequestException.invalidResponseMode(message: message ?? "")
        case "UnsupportedDidUrl":
            exception = DidResolverExceptions.unsupportedDidUrl(message: message)
        case "DidResultionFailed":
            exception = DidResolverExceptions.didResolutionFailed(message: message)
        case "PublicKeyResolutionFailed":
            exception = JwtVerificationException.publicKeyResolutionFailed(message: message ?? "Error occurred while resolve public key")
        default:
            exception = AuthorizationRequestException.unexpectedError(message: "An unexpected exception occurred: exception type: \(exceptionType)")
        }
        error(getLogTag(className), exception)
        return exception
    }
}
