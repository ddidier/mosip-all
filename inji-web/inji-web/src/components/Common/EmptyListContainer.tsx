import React from "react";
import {EmptyListContainerProps} from "../../types/components";

export const EmptyListContainer: React.FC<EmptyListContainerProps> = ({content}) => {
    return <React.Fragment>
        <div data-testid="EmptyList-Outer-Container"
             className="flex justify-center items-center w-full mx-auto bg-iw-background
             my-auto flex-col h-72 shadow-md shadow-iw-shadow border-t-iw-shadow px-10">
            <p data-testid="EmptyList-Text"
               className="text-center text-light-title dark:text-dark-title">{content}</p>
        </div>
    </React.Fragment>
}
