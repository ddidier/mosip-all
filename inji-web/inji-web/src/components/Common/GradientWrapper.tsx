import React from "react";
import { generateRandomString } from "../../utils/misc";

export const GradientWrapper: React.FC<BackGroundImageProps> = ({children}) => {
    const uniqueId = `blue-gradient-${
        generateRandomString(7,'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789')
    }`;
    const childrenWithProps = React.Children.map(children, (child) => {
        if (React.isValidElement(child)) {
            // @ts-ignore
            return React.cloneElement(child, {style: { fill: `url(#${uniqueId})`,}});
        }
        return child;
    });

    return <React.Fragment>
        <svg width="0" height="0">
                <linearGradient id={uniqueId} x1="0%" y1="0%" x2="100%" y2="100%">
                    <stop stopColor="var(--iw-color-primary)" offset="0%" />
                    <stop stopColor="var(--iw-color-secondary)" offset="100%" />
                </linearGradient>
        </svg>
        {childrenWithProps}
    </React.Fragment>
};

export type BackGroundImageProps = {
    children: React.ReactNode;
}