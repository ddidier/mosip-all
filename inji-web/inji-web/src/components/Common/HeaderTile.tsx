import React from "react";
import {HeaderTileProps} from "../../types/components";

export const HeaderTile: React.FC<HeaderTileProps> = ({content, subContent}) => {
    return <React.Fragment>
        <div data-testid="HeaderTile-Text"
             className={"font-bold text-iw-title mt-8 text-xl flex"}>{content}</div>
        <div data-testid="HeaderTile-Text-SubContent"
             className={"text-iw-subTitle text-md mb-8 mt-2 flex"}>{subContent}</div>
    </React.Fragment>
}
