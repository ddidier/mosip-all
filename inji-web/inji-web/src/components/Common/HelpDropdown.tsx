import React, {useState, useTransition} from "react";
import {FaCheck} from "react-icons/fa6";
import {RiArrowDownSFill, RiArrowUpSFill} from "react-icons/ri";
import {GradientWrapper} from "./GradientWrapper";
import {useNavigate} from "react-router-dom";
import {renderGradientText} from "../../utils/builder";
import { useTranslation } from "react-i18next";

export const HelpDropdown: React.FC = () => {
    const navigate = useNavigate();
    const [isOpen, setIsOpen] = useState(false);
    const {t}=useTranslation("HelpDropdown");

    return <div className={"flex flex-row justify-center items-center"}
                data-testid={"HelpDropdown-Outer-Div"}
                onBlur={()=>setIsOpen(false)}
                tabIndex={0}
                role="button">

        <div className="relative inline-block ms-1">
            <button
                type="button"
                className="inline-flex items-center"
                data-testid={"HelpDropdown-Button"}
                onMouseDown={() => setIsOpen(open => !isOpen)}>
                    <p data-testid={`Selected-DropDown`}>
                        {t("heading")}
                </p>
                {isOpen ?
                    <GradientWrapper>
                        <RiArrowUpSFill size={20} color={'var(--iw-color-languageArrowIcon)'} />
                    </GradientWrapper> :
                    <GradientWrapper>
                        <RiArrowDownSFill size={20} color={'var(--iw-color-languageArrowIcon)'}/>
                    </GradientWrapper> }
            </button>

            {isOpen && (
                <div
                    className="absolute w-40 right-0 mt-3 rounded-md shadow-lg bg-iw-background overflow-hidden font-normal">
                    <ul className="py-1 divide-y divide-gray-200">
                        <li data-testid={`Help-DropDown-Item`}>
                            <button
                                type="button"
                                className="w-full px-4 py-2 text-left text-sm font-semibold hover:bg-gray-100 flex items-center justify-between flex-row"
                                onMouseDown={() => {setIsOpen(open=>!open); navigate("/help");}}>{t("item1")}
                            </button>
                        </li>
                    </ul>
                </div>
            )}
        </div>
    </div>
}
