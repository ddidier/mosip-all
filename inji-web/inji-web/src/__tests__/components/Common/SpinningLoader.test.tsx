import React from 'react';
import { render} from '@testing-library/react';
import {SpinningLoader} from "../../../components/Common/SpinningLoader";



describe("Testing the Layout of Spinning Loader Container",() => {
    test('Check if the layout is matching with the snapshots', () => {
        const {asFragment} = render(<SpinningLoader />)
        expect(asFragment()).toMatchSnapshot();
    });
})

