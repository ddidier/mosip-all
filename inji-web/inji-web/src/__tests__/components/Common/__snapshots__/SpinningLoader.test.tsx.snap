// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Layout of Spinning Loader Container Check if the layout is matching with the snapshots 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center"
    data-testid="SpinningLoader-Container"
  >
    <div
      aria-busy="true"
      aria-label="oval-loading"
      class="sc-beySPh gZmPOF"
      data-testid="oval-loading"
      role="progressbar"
    >
      <svg
        data-testid="oval-svg"
        height="80"
        stroke="var(--iw-color-spinningLoaderPrimary)"
        viewBox="-20.5 -20.5 43 43"
        width="80"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <g
            data-testid="oval-secondary-group"
            stroke-width="3"
            transform="translate(1 1)"
          >
            <circle
              cx="0"
              cy="0"
              r="20"
              stroke="var(--iw-color-spinningLoaderSecondary)"
              stroke-opacity=".5"
              stroke-width="3"
            />
            <path
              d="M20 0c0-9.94-8.06-20-20-20"
            >
              <animatetransform
                attributeName="transform"
                dur="1s"
                from="0 0 0"
                repeatCount="indefinite"
                to="360 0 0"
                type="rotate"
              />
            </path>
          </g>
        </g>
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
